src/uwtools/drivers/shave.py

```diff
@@ -16,13 +19,34 @@ class Shave(DriverTimeInvariant):
 
     # Workflow tasks
 
+    @task
+    def input_config_file(self):
+        """
+        The input config file.
+        """
+        path = self._input_config_path
+        yield self.taskname(str(path))
+        yield asset(path, path.is_file)
+        config = self.config["config"]
+        input_file = Path(config["input_grid_file"])
+        yield asset(input_file, input_file.is_file)
+        flags = [
+            config[key] for key in ["nx", "ny", "nhalo", "input_grid_file", "output_grid_file"]
+        ]
+        content = "{} {} {} '{}' '{}'".format(*flags)
+        with writable(path) as f:
+            print(content, file=f)
```

Comment:
```suggestion
        with open(path, "w", encoding="utf-8") as f:
            print(content, file=f)
```

Reply:
@maddenp-noaa I applied this change and realized that `writable` does the helpful additional task of ensuring the directory where the file is meant to land exists.

We use it in the base class in the definition of `_write_runscript` and it's called as part of `dump` methods when we're creating supported config formats. I'm not sure whether I should just leave the `writable` call here (and above) because it is consistent, or whether I should have a `makedirs` step explicitly of this level of the drivers where it doesn't appear in the others.

Reply:
I think use of `writable()` in the `dump` methods makes sense because we might be dumping those to a file or to `stdout`. Its use in `_write_runscript()` is questionable, as that _must_ be a file (because we set the executable bit on the file immediate after).

But maybe I'm being too strict here: As long as there's no chance that the `path` argument to `writable()` is `None`, it should be safe to use. That's probably the case here, so please feel free to resolve this conversation.

Reply:
How bad would it be to add a line with `assert path is not None`?

Reply:
Also, I'm seeing now that we are internally defining a path that is not None with this class.

Reply:
Yeah, I can't see a possibility of a `None` value. I think it's fine.